packages/yew-agent/src/oneshot/provider.rs

```diff
@@ -0,0 +1,129 @@
+use core::fmt;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use serde::{Deserialize, Serialize};
+use yew::prelude::*;
+
+use super::{Oneshot, OneshotBridge, OneshotSpawner};
+use crate::utils::get_next_id;
+use crate::worker::WorkerProviderProps;
+use crate::{Bincode, Codec, Reach};
+
+pub(crate) struct OneshotProviderState<T>
+where
+    T: Oneshot + 'static,
+{
+    id: usize,
+    spawn_bridge_fn: Rc<dyn Fn() -> OneshotBridge<T>>,
+    reach: Reach,
+    held_bridge: Rc<RefCell<Option<OneshotBridge<T>>>>,
+}
+
+impl<T> fmt::Debug for OneshotProviderState<T>
+where
+    T: Oneshot,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("OneshotProviderState<_>")
```

Comment:
The `_` can be [`std::any::type_name`](https://doc.rust-lang.org/stable/std/any/fn.type_name.html) of `T`